src/components/core/PivotTable.tsx

```diff
@@ -866,6 +891,32 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
         );
     }
 
+    private setGroupingProvider(sortedByFirstAttr: boolean) {
+        this.groupingProvider = GroupingProviderFactory.createProvider(sortedByFirstAttr);
+    }
+
+    private sortedByFirstAttribute(columnDefs: ColDef[]) {
+        const sortedColumnIndexes: ISortedByColumnIndexes = columnDefs.reduce(
+            (
+                sortStack: ISortedByColumnIndexes,
+                columnDef: ColDef,
+                columnIndex: number
+            ) => {
+                if (columnDef.sort) {
+                    sortStack.all.push(columnIndex);
+                }
+                if (columnDef.type === ROW_ATTRIBUTE_COLUMN && columnDef.sort) {
```

Comment:
it could be optimised right ? 
```
if (columnDef.sort) {
	sortStack.all.push(columnIndex);

	if (columnDef.type === ROW_ATTRIBUTE_COLUMN) {
		sortStack.attributes.push(columnIndex);
	}
}               
```